### PR TITLE
Switch linting to ESLint and Prettier

### LINKED ISSUE #97: Stick with JSCS or switch to another linting tool
JSCS has been merged with ESLint, and development on JSCS has stopped.

@MoOx I thought I remembered seeing somewhere that you said you wanted to switch to Prettier, but I can't find it now.

Should we switch to ESLint or Prettier? Or, since for now JSCS works fine, should we stick with that?

### PR DESCRIPTION
**Pull request by BehindTheMath**
Close #97.

### CODE REVIEW

**Comment by BehindTheMath** [comment]
I'm not sure why Travis is failing now. [`.eslintignore`](https://github.com/MoOx/pjax/blob/79295232e7b82879b96edeb9f12d67f7c7031e1f/.eslintignore) excludes `node_modules/**`.

File: package.json [hunk 1]
```diff
@@ -28,21 +28,24 @@
...
+    "prettier": "^1.14.3",
     "serve": "^6.4.4",
     "tap-nyc": "^1.0.3",
     "tap-spec": "^4.1.1",
     "tape": "^4.8.0",
     "uglify-js": "^3.3.8"
   },
   "scripts": {
-    "lint": "jscs . && jshint . --exclude-path .gitignore",
+    "lint": "eslint **/*.js",
```
Comment (MoOx) [inline package.json]: I think you can just do `"eslint ."`

File: package.json [hunk 2]
```diff
@@ -53,5 +56,17 @@
...
     "release": "npmpub"
+  },
+  "husky": {
+    "hooks": {
+      "pre-commit": "npm test && lint-staged"
+    }
+  },
+  "lint-staged": {
+    "*.js": [
+      "eslint --fix **/*.js",
```
Comment (MoOx) [inline package.json]: File will be passed as argument, so you don't need to send `**/*.js`. Just `"eslint --fix",

File: .eslintignore [hunk 3]
```diff
@@ -0,0 +1,4 @@
+pjax.js
+node_modules/**
```
Comment (MoOx) [inline .eslintignore]: No need for this. In 
https://eslint.org/docs/user-guide/configuring#eslintignore
you can read

> # /node_modules/* and /bower_components/* in the project root are ignored by default

I think it's the same for prettier.

Something I do sometimes is to symlink this *ignore to .gitignore (eg: that's what I do here https://github.com/MoOx/eslint-config-i-am-meticulous)

**Comment by BehindTheMath** [comment]
@MoOx How about now?

**Comment by MoOx** [comment]
Looks good!

**Pull request merged**